### PR TITLE
AnchorFM: Augment Signup Events with Podcasting Site Information

### PR DESCRIPTION
**Pull request by krymson24**
#### Changes proposed in this Pull Request

Adds a `is_podcasting_site` event prop to the `calypso_newsite_start` and `calypso_signup_start` event to answer the question "How many podcast sites were launched?". Context for this approach to support funneling: p1608037471216200-slack-C0Q664T29

Screenshots
Calypso Signup Start
Screen Shot 2020-12-15 at 9 20 48 PM
Calypso Newsite Start
Screen Shot 2020-12-15 at 9 20 02 PM

#### Testing instructions

1. Go to http://calypso.localhost:3000/new?anchor_podcast={podcast_id}
2. After 5 minutes go to MC -- `/tracks/live/?eventname=calypso_signup_start` and ` `/tracks/live/?eventname=calypso_signup_start` look for your specific signup (the username isn't captured so it could be difficult to find) and ensure there is a the `is_podcasting_site` prop

Fixes 374-gh-Automattic/dotcom-manage

### CODE REVIEW

**Review by mmtr — APPROVED** [review]
Nice, it works great! I also verified the `is_podcasting_site` prop is set to false for regular Gutenboarding flows (those without a `anchor_podcast` param).

**Comment by mmtr** [comment]
Hold off before merging this, @krymson24. I think @kwight wants us to explore an alternative approach that defines a custom `anchor-fm` flow to be used when calling `trackEventWithFlow` (3rd param) instead of using the default `gutenboarding` flow:

https://github.com/Automattic/wp-calypso/blob/b0dee8c85ef7a11fdf98092ea2c385a32dc950bf/client/landing/gutenboarding/lib/analytics/index.ts#L22

The good thing about that approach is that we wouldn't pollute the stats for the existing funnels measuring the regular Gutenboarding flow (signup completion rate may be different between regular sites and podcasting sites), but we need to make sure there are no unexpected side effects of doing that. cc @allilevine @mreishus

File: client/landing/gutenboarding/lib/analytics/index.ts
Comment (mmtr) [inline client/landing/gutenboarding/lib/analytics/index.ts]: The `trackEventWithFlow` function accepts a third param with the flow identifier. Is it possible to use it?

Comment (krymson24) [reply client/landing/gutenboarding/lib/analytics/index.ts]: I initially tried, and didn't see it come through, but I tested it again, and the value is coming through. It might have been perhaps I couldn't find it the first time around.

Thanks!

**Comment by kwight** [comment]
> Hold off before merging this, @krymson24. I think @kwight wants us to explore an alternative approach that defines a custom anchor-fm flow to be used when calling trackEventWithFlow (3rd param) instead of using the default gutenboarding flow

We're cleared to try this alternative approach (p7rd6c-2DP-p2), so this PR can be put aside.

**Pull request closed without merging**
